### PR TITLE
CTCP-4547:CREATE PACKAGESHIPPINGMARKSPAGE

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

File: app/config/Constants.scala [hunk 1]
```diff
@@ -33,6 +33,9 @@ object Constants {
 
   val Other = "D"
 
+  lazy val maxItemDescriptionLength: Int     = 512
+  lazy val maxPackageShippingMarkLength: Int = 512
+
```
Comment (justega247) [inline app/config/Constants.scala:38]: These can be moved to the Constants file in the forms directory

File: test/forms/PackageShippingMarkFormProviderSpec.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import play.api.i18n.Lang.defaultLang
+import play.api.i18n.{Messages, MessagesApi, MessagesImpl}
+import play.api.inject.guice.GuiceApplicationBuilder
+import viewModels.PackageShippingMarksViewModel
+
+class PackageShippingMarkFormProviderSpec extends StringFieldBehaviours {
+
+  private val invalidKey = "houseConsignment.item.packageShippingMark.error.invalid"
+  private val hcIndex    = Index(0)
+  private val itemIndex  = Index(0)
```
Comment (justega247) [inline test/forms/PackageShippingMarkFormProviderSpec.scala:35]: You can probably just extend with SpecBase so you don't have to redefine these.

File: test/pages/houseConsignment/index/items/PackageShippingMarkPageSpec.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ * limitations under the License.
+ */
+
+package pages.houseConsignment.index.items
+
+import pages.behaviours.PageBehaviours
+
+class PackageShippingMarkPageSpec extends PageBehaviours {
+
+  "CommodityCodePage" - {
```
Comment (justega247) [inline test/pages/houseConsignment/index/items/PackageShippingMarkPageSpec.scala:23]: This should be PackageShippingMarkPage I believe.

File: app/pages/houseConsignment/index/items/PackageShippingMarkPage.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+import models.Index
+import pages.QuestionPage
+import pages.sections.ItemsSection
+import play.api.libs.json.JsPath
+
+case class PackageShippingMarkPage(houseConsignment: Index, itemIndex: Index, packageIndex: Index) extends QuestionPage[String] {
+
+  override def path: JsPath = ItemsSection(houseConsignment).path \ itemIndex.position \ "Packaging" \ packageIndex.position \ toString
+
+  override def toString: String = "packageShippingMarkIndex"
```
Comment (jackheslop96) [inline app/pages/houseConsignment/index/items/PackageShippingMarkPage.scala:28]: This should be `shippingMarks`

File: app/viewModels/PackageShippingMarksViewModel.scala [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+ */
+
+package viewModels
+
+import models.{CheckMode, Index, Mode, NormalMode}
+import play.api.i18n.Messages
+
+import javax.inject.Inject
+
+case class PackageShippingMarksViewModel(houseConsignmentIndex: Index, itemIndex: Index, mode: Mode) {
```
Comment (jackheslop96) [inline app/viewModels/PackageShippingMarksViewModel.scala:24]: Let's press pause on these view models until we've got a structure we're happy with. This looks to be copied from Jake's branch but there are changes pending.

File: conf/messages.en [hunk 6]
```diff
@@ -393,7 +393,6 @@ cannotSendUnloadingRemarks.linkText = View the status of this notification
 departureTransportMeans.identification.heading=Which identification do you want to use for the new departure means of transport?
 departureTransportMeans.identification.title=Which identification do you want to use for the new departure means of transport?
 departureTransportMeans.identification.error.required=Select which identification you want to use for the new departure means of transport
-departureTransportMeans.identification.paragraph=This is the means of transport used from the UK office of departure to a UK port or airport.
```
Comment (jackheslop96) [inline conf/messages.en:396]: Were you meant to remove this? Perhaps a merge conflict gone wrong?

Comment (jackheslop96) [reply conf/messages.en:396]: Looks like `TransportMeansIdentificationViewSpec` is missing a check to see if this paragraph is there. Would you be able to add that in?

**Review by justega247 — DISMISSED** [review]

**Pull request merged**